cmake error: define_property not given a BRIEF_DOCS <brief-doc> argument.

After commit 0fa0e4195828751cff5fe51a2c37dfd003cd61db, I started getting this error in my builds:

```
CMake Error at /home/user/workspace/external/zephyr/cmake/linker/linker_flags_template.cmake:61 (define_property):
  define_property not given a BRIEF_DOCS <brief-doc> argument.
Call Stack (most recent call first):
  /home/user/workspace/external/zephyr/cmake/target_toolchain_flags.cmake:37 (include)
  /home/user/workspace/external/zephyr/cmake/modules/kernel.cmake:153 (include)
  /home/user/workspace/external/zephyr/cmake/modules/zephyr_default.cmake:138 (include)
  /home/user/workspace/external/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/user/workspace/external/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:3 (find_package)


CMake Error at /home/user/workspace/external/zephyr/cmake/linker/linker_flags_template.cmake:62 (define_property):
  define_property not given a FULL_DOCS <full-doc> argument.
Call Stack (most recent call first):
  /home/user/workspace/external/zephyr/cmake/target_toolchain_flags.cmake:37 (include)
  /home/user/workspace/external/zephyr/cmake/modules/kernel.cmake:153 (include)
  /home/user/workspace/external/zephyr/cmake/modules/zephyr_default.cmake:138 (include)
  /home/user/workspace/external/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/user/workspace/external/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:3 (find_package)


CMake Error at /home/user/workspace/external/zephyr/cmake/linker/linker_flags_template.cmake:63 (define_property):
  define_property not given a FULL_DOCS <full-doc> argument.
Call Stack (most recent call first):
  /home/user/workspace/external/zephyr/cmake/target_toolchain_flags.cmake:37 (include)
  /home/user/workspace/external/zephyr/cmake/modules/kernel.cmake:153 (include)
  /home/user/workspace/external/zephyr/cmake/modules/zephyr_default.cmake:138 (include)
  /home/user/workspace/external/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/user/workspace/external/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:3 (find_package)


CMake Error at /home/user/workspace/external/zephyr/cmake/linker/linker_flags_template.cmake:64 (define_property):
  define_property not given a FULL_DOCS <full-doc> argument.
Call Stack (most recent call first):
  /home/user/workspace/external/zephyr/cmake/target_toolchain_flags.cmake:37 (include)
  /home/user/workspace/external/zephyr/cmake/modules/kernel.cmake:153 (include)
  /home/user/workspace/external/zephyr/cmake/modules/zephyr_default.cmake:138 (include)
  /home/user/workspace/external/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/user/workspace/external/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:3 (find_package)


CMake Error at /home/user/workspace/external/zephyr/cmake/linker/linker_flags_template.cmake:65 (define_property):
  define_property not given a FULL_DOCS <full-doc> argument.
Call Stack (most recent call first):
  /home/user/workspace/external/zephyr/cmake/target_toolchain_flags.cmake:37 (include)
  /home/user/workspace/external/zephyr/cmake/modules/kernel.cmake:153 (include)
  /home/user/workspace/external/zephyr/cmake/modules/zephyr_default.cmake:138 (include)
  /home/user/workspace/external/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/user/workspace/external/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:3 (find_package)
```

_Originally posted by @matinlotfali in https://github.com/zephyrproject-rtos/zephyr/issues/103105#issuecomment-3832653241_

--- a/cmake/linker/linker_flags_template.cmake
+++ b/cmake/linker/linker_flags_template.cmake
@@ -58,8 +58,10 @@ set_property(TARGET linker PROPERTY lto_arguments_st)
 # front-end for ld.
 set_compiler_property(PROPERTY specs)
 
-define_property(TARGET PROPERTY no_optimization INHERITED FULL_DOCS "INHERIT")
-define_property(TARGET PROPERTY optimization_debug INHERITED BRIEF_DOCS "INHERIT")
-define_property(TARGET PROPERTY optimization_speed INHERITED BRIEF_DOCS "INHERIT")
-define_property(TARGET PROPERTY optimization_size INHERITED BRIEF_DOCS "INHERIT")
-define_property(TARGET PROPERTY optimization_size_aggressive INHERITED BRIEF_DOCS "INHERIT")
+# The optimization properties are used for controlling inheritance for linker settings.
+set(linker_property_docs BRIEF_DOCS "INHERIT" FULL_DOCS "Linker property with inheritance")
+define_property(TARGET PROPERTY no_optimization INHERITED ${linker_property_docs})
+define_property(TARGET PROPERTY optimization_debug INHERITED ${linker_property_docs})
+define_property(TARGET PROPERTY optimization_speed INHERITED ${linker_property_docs})
+define_property(TARGET PROPERTY optimization_size INHERITED ${linker_property_docs})
+define_property(TARGET PROPERTY optimization_size_aggressive INHERITED ${linker_property_docs})

--- a/cmake/modules/hwm_v2.cmake
+++ b/cmake/modules/hwm_v2.cmake
@@ -30,7 +30,7 @@ function(kconfig_gen bin_dir file dirs comment)
   endforeach()
 
   file(WRITE ${kconfig_file}.tmp "${kconfig_output}")
-  file(COPY_FILE ${kconfig_file}.tmp ${kconfig_file} ONLY_IF_DIFFERENT)
+  zephyr_file_copy(${kconfig_file}.tmp ${kconfig_file} ONLY_IF_DIFFERENT)
 endfunction()
 
 # 'SOC_ROOT' and 'ARCH_ROOT' are prioritized lists of directories where their

